src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -256,43 +256,57 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
       this.search = '';
     }
     const columns = this.columns.filter(c => c.cellTransformation !== CellTemplate.sparkline);
+    if (this.search.match(/['"][^'"]+['"]/)) {
+      this.search = this.search.replace(/['"][^'"]+['"]/g, (match: string) => {
+        return match.replace(/(['"])([^'"]+)(['"])/, '$2').replace(' ', '+');
+      });
+    }
     // update the rows
-    this.rows = this.subSearch(this.data, this.search.toLowerCase().split(/[, ]/), columns);
+    this.rows = this.subSearch(this.data, this.search.toLowerCase().split(/[ ]/), columns);
     // Whenever the filter changes, always go back to the first page
     this.table.offset = 0;
   }
 
   subSearch (data: any[], currentSearch: string[], columns: CdTableColumn[]) {
-    let tempColumns: CdTableColumn[];
     if (currentSearch.length === 0 || data.length === 0) {
       return data;
     }
-    const searchWords: string[] = currentSearch.pop().split(':');
-    if (searchWords.length === 2) {
-      tempColumns = [...columns];
-      columns = columns.filter((c) => c.name.toLowerCase().indexOf(searchWords[0]) !== -1);
+    const searchTerms: string[] = currentSearch.pop().split(':');
```

Comment:
you could replace `+` with ` `, after `pop()`. This way you don't need to define `getSearchTerm`.

Reply:
good spotting :)